### PR TITLE
feat(marked): update tokenizer api to 2.x

### PR DESCRIPTION
**Pull request by AviVahl**
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
https://github.com/markedjs/marked/blob/master/docs/USING_PRO.md#block-level-tokenizer-methods
https://github.com/markedjs/marked/compare/v1.2.9...master
https://github.com/markedjs/marked/releases/tag/v2.0.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

also applied prettier to definition.

### CODE REVIEW

File: types/marked/index.d.ts [hunk 1]
```diff
@@ -137,7 +145,7 @@ declare namespace marked {
         constructor(options?: MarkedOptions);
         options: MarkedOptions;
         space(src: string): Tokens.Space;
-        code(src: string, token: Token): Tokens.Code;
+        code(src: string): Tokens.Code;
```
Comment (AviVahl) [inline types/marked/index.d.ts:148]: actual change

File: types/marked/index.d.ts [hunk 2]
```diff
@@ -153,9 +161,8 @@ declare namespace marked {
         escape(src: string): Tokens.Escape;
         tag(src: string, inLink: boolean, inRawBlock: boolean): Tokens.Tag;
         link(src: string): Tokens.Image | Tokens.Link;
-        reflink(src: string, links: Tokens.Link[] |Tokens.Image[]): Tokens.Link |Tokens.Image | Tokens.Text;
-        strong(src: string): Tokens.Strong;
-        em(src: string): Tokens.Em;
+        reflink(src: string, links: Tokens.Link[] | Tokens.Image[]): Tokens.Link | Tokens.Image | Tokens.Text;
+        emStrong(src: string, maskedSrc: string, prevChar: string): Tokens.Em | Tokens.Strong;
```
Comment (AviVahl) [inline types/marked/index.d.ts:165]: actual change

File: types/marked/marked-tests.ts [hunk 3]
```diff
@@ -1,5 +1,15 @@
 import * as marked from 'marked';
 
+const tokenizer = new marked.Tokenizer();
+
+tokenizer.emStrong = (src, _maskedSrc, _prevChar) => {
```
Comment (AviVahl) [inline types/marked/marked-tests.ts:5]: added test

**Review by sarunint — APPROVED** [review]
LGTM

**Comment by sarunint** [comment]
Ready to merge

**Comment by AviVahl** [comment]
Thanks for the review. Ready to merge.

**Pull request merged**